enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,483 @@
+|Cloud platform SNO no workers           |Cloud   |0  |0  |SR |SR |W  |M  |No                                               |No                                            |Works, unsupported, tested          |
+|None platform SNO day1 single worker    |None    |1  |1  |-  |SR |-  |M  |Yes, ingress floats                              |No, ingress pinned to master pool             |Impossible to install               |
+|Cloud platform SNO day1 single worker   |Cloud   |1  |1  |SR |HA |W  |W  |Yes - provided, ingress floats                   |Yes - provided, there are two ingress replicas|Works, unsupported, untested        |
+|None platform SNO day1 multiple workers |None    |2+ |2+ |-  |SR |-  |M  |Yes, ingress is on two different nodes           |No, one ingress and is pinned to master pool  |Impossible to install               |
+|Cloud platform SNO day1 multiple workers|Cloud   |2+ |2+ |HA |HA |W  |W  |Yes - provided, ingress is on two different nodes|Yes - provided, there are two ingress replicas|Works, unsupported, untested        |
+|None platform SNO only day2 workers     |None    |0  |1+ |SR |SR |W  |M  |Yes, ingress floats                              |No, ingress pinned to master pool             |Ingress issue, unsupported, untested|
+|Cloud platform SNO only day2 workers    |Cloud   |0  |1+ |SR |SR |W  |M  |Yes - provided, ingress floats                   |Yes - provided, ingress floats                |Works, unsupported, untested        |
+
+Some notes about this table and the decisions made in it:
+
+The table assumes that the "Infrastrcture Topology" value is read only. That
+means an optimal value for this parameter must be determined during
+installation. You can see the consequences of that, for example, in the last
+row of the table - even though we add workers in day 2 to the cluster, the
+"Infrastrcture Topology" stays "SingleReplica" even though that cluster could
+benefit from the additional worker nodes for the purposes of highly-available
+infrastructure (as it's running in the cloud, so it has a load-balancer). In
+the future it may be worth revisiting making this value configurable.
+
+The "None platform SNO day1 single/multiple worker" scenario rows aren't really
+possible today as there is no real way to install this kind of cluster. That's
+why their CIT / CTP have been left empty.
+
+It's proposed that the only scenarios in which the "Infrastrcture Topology"
+parameter will be "HighlyAvailable" are when installing a single control-plane
+node cluster in the cloud with one or more day-1 workers. Cloud deployments
+have a load balancer so they can benefit from having highly-available ingress.
+
+Today, if you only have 1 day-1 worker in a single control-plane node cluster
+in the cloud, then the "Infrastrcture Topology" is set to "SingleReplica" - I
+don't believe there's a good reason for that, as both the control-plane node
+and the worker node are actually in the worker pool (TODO: make sure this is
```

Comment:
Why is the control-plane node in the worker pool in this scenario? The installer would not mark the masters as schedulable. Is there something happening in-cluster to make the masters schedulable?

Reply:
Yeah it was just some assumption I based this paragraph on that I later realized might likely be completely wrong, this is why the TODO is there - I wanted to try it in practice, haven't got around to it yet.